portfolio/src/main/webapp/script.js

```diff
@@ -193,6 +293,189 @@ function circularIndex(n, items) {
+
+  for (let icon of optionIcons) {
+    icon.className = icon.className.replace('fas', 'far');
+  }
+
+  let highlightIcon;
+
+  switch (reaction) {
+    case reacts.like:
+      highlightIcon = document.getElementById('likeBtn');
+      break;
+    case reacts.love:
+      highlightIcon = document.getElementById('loveBtn');
+      break;
+    case reacts.wow:
+      highlightIcon = document.getElementById('wowBtn');
+      break;
+    case reacts.laugh:
+      highlightIcon = document.getElementById('laughBtn');
+      break;
+    default:
+      return;
+  }
+
+  highlightIcon.className = reaction.solidIcon;
+}
+
+function generateReactCounter(container, reaction, count) {
+  const box = document.createElement('div');
+  box.className = 'curr-react';
+  
+  const icon = document.createElement('i');
+  icon.className = reaction.solidIcon;
+  icon.style.color = reaction.colour;
+
+  const countDiv = document.createElement('div');
+  countDiv.innerText = count.toString();
+
+  box.appendChild(icon);
+  box.appendChild(countDiv);
+  container.appendChild(box);
+}
+
+function updatePostReacts(reactCounts) {
+  const reactCounter = document.getElementsByClassName("reacts-container")[0];
+  reactCounter.innerHTML = '';
+
+  if (reactCounts.likeCount > 0) {
+    generateReactCounter(reactCounter, reacts.like, reactCounts.likeCount);
+  }
+  if (reactCounts.loveCount > 0) {
+    generateReactCounter(reactCounter, reacts.love, reactCounts.loveCount);
+  }
+  if (reactCounts.wowCount > 0) {
+    generateReactCounter(reactCounter, reacts.wow, reactCounts.wowCount);
+  }
+  if (reactCounts.laughCount > 0) {
+    generateReactCounter(reactCounter, reacts.laugh, reactCounts.laughCount);
+  }
+}
+
+/**
+ * Fetch reactions on the given post from backend.
+ * @param {string} postId - identifier for post.
+ */
+function fetchPostReacts(postId) {
+  const postParam = new URLSearchParams();
+  postParam.append('postId', postId);
+
+  fetch(postReactionsURL + '?' + postParam.toString())
+  .then((response) => response.json())
+  .then(data => {
+    updatePostReacts(data);
+  });
+}
+
+/**
+ * Function to update the reactions on the post on backend.
+ * @param oldReact - reaction that current user previously selected.
+ * @param newReact - new reaction current user now selected.
+ */
+function newPostReaction(oldReact, newReact) {
+  const data = new URLSearchParams();
+  data.append('postId', picData[currIndex].id);
+
+  if (oldReact !== reacts.none) {
+    data.append('oldReact', oldReact.id);
+  }
+
+  if (newReact !== reacts.none) {
+    data.append('newReact', newReact.id);
+  }
+
+  const options = {
+    headers: {
+      'Content-Type': 'application/x-www-form-urlencoded'
+    },
+    method: 'POST',
+    body: data.toString()
+  };
+
+  fetch(postReactionsURL, options)
+  .then((response) => {
+    if (response.status === STATUS_OK) {
+      fetchPostReacts(picData[currIndex].id);
+    }
+  });
+}
+
+/**
```

Comment:
I don't understand how this works. Can there be 2 different reactions at the same time?

Reply:
So the concept is that there can already be an existing reaction a user has made on a post, say a 'Wow' react, but they could click the 'Like' reaction instead, so the reaction would need to switch from wow to like. On the backend, this would mean needing to decrement the number of 'wow' reactions and incrementing the 'likes'. So that is why two reactions can be sent to the backend.

Reply:
If only one reaction is possible per post, why do you need to store count of wows, count of likes separately?

Reply:
It is one reaction per post per user, so several users can react on an image with different reactions. Sort of modelled similar to LinkedIn or Facebook's reactions on posts.

Reply:
At the moment there is no authentication about which user is sending a reaction, so it is session based. But I am working on some authentication stuff to link a reaction specifically to an account so someone can't spam reactions by refreshing.

Reply:
But then the data model should store which user did which reaction on which post. Now it stores only number of each reaction.

Reply:
So this was part of last week's work, in which there was no authentication, or method of identifying a user. The user-specific data model is what I am working on now with comments and reactions.

Reply:
Looks like this PR depends on #14 then